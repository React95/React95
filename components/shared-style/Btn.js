import styled from 'styled-components';

export default styled.button`
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;

  font-size: 12px;

  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,
    1px 1px 0 0px #000;

  &:disabled {
    color: #868a8e;
  }

  &:focus {
    outline: none;
    box-shadow: inset 1px 1px 0px 1px #ffffff,
      inset -0.5px -0.5px 0px 1px #868a8e, 1px 1px 0 1px #000;
    outline: 1px dotted #000;
    outline-offset: -5px;
  }

  &:active {
    padding: 8px 20px 5px;

    outline: 1px dotted #000;
    outline-offset: -5px;

    box-shadow: inset 0 0 0 1px #868a8e, 0 0 0 1px #000;
  }
`;
