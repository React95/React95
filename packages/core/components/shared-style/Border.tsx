import {
  css,
  DefaultTheme,
  FlattenInterpolation,
  ThemeProps,
} from 'styled-components';

interface IBorderStyle {
  direction?: 'extrude' | 'intrude';
  useBoxShadow?: boolean;
}

interface IBorder extends IBorderStyle {
  outerTopLeft: FlattenInterpolation<ThemeProps<DefaultTheme>>;
  innerTopLeft: FlattenInterpolation<ThemeProps<DefaultTheme>>;
  outerBottomRight: FlattenInterpolation<ThemeProps<DefaultTheme>>;
  innerBottomRight: FlattenInterpolation<ThemeProps<DefaultTheme>>;
}

export const createBorder = ({
  direction,
  outerTopLeft,
  innerTopLeft,
  outerBottomRight,
  innerBottomRight,
}: IBorder) => css`
  position: relative;
  border-width: 2px;
  border-style: solid;
  border-top-color: ${direction == 'extrude' ? outerTopLeft : innerBottomRight};
  border-right-color: ${direction == 'extrude'
    ? outerBottomRight
    : outerTopLeft};
  border-bottom-color: ${direction == 'extrude'
    ? outerBottomRight
    : outerTopLeft};
  border-left-color: ${direction == 'extrude'
    ? outerTopLeft
    : innerBottomRight};
  :before {
    content: '';
    position: absolute;
    left: -1px;
    top: -1px;
    width: 100%;
    height: 100%;
    border-style: solid;
    border-width: 1px;
    border-top-color: ${direction == 'extrude'
      ? innerTopLeft
      : outerBottomRight};
    border-right-color: ${direction == 'extrude'
      ? innerBottomRight
      : innerTopLeft};
    border-bottom-color: ${direction == 'extrude'
      ? innerBottomRight
      : innerTopLeft};
    border-left-color: ${direction == 'extrude'
      ? innerTopLeft
      : outerBottomRight};
    pointer-events: none;
  }
`;

export const createBoxShadowBorder = ({
  direction,
  outerTopLeft,
  innerTopLeft,
  outerBottomRight,
  innerBottomRight,
}: IBorder) => css`
  position: relative;
  border-width: 1px;
  border-style: solid;
  border-top-color: ${direction == 'extrude' ? outerTopLeft : innerBottomRight};
  border-right-color: ${direction == 'extrude'
    ? outerBottomRight
    : outerTopLeft};
  border-bottom-color: ${direction == 'extrude'
    ? outerBottomRight
    : outerTopLeft};
  border-left-color: ${direction == 'extrude'
    ? outerTopLeft
    : innerBottomRight};
  box-shadow: inset -1px -1px 0 0px
      ${direction == 'extrude' ? innerBottomRight : innerTopLeft},
    inset 1px 1px 0 0px
      ${direction == 'extrude' ? innerTopLeft : outerBottomRight};
`;

/*

    box-shadow: inset -1px -1px 0 0px
        ${direction == 'extrude' ? outerBottomRight : outerTopLeft},
      inset 1px 1px 0 0px
        ${direction == 'extrude' ? outerTopLeft : innerBottomRight},
      inset -2px -2px 0 0px
        ${direction == 'extrude' ? innerBottomRight : innerTopLeft},
      inset 2px 2px 0 0px
        ${direction == 'extrude' ? innerTopLeft : outerBottomRight};

*/

export const border = ({
  direction = 'extrude',
  useBoxShadow = false,
}: IBorderStyle = {}) => {
  let borderFn = useBoxShadow ? createBoxShadowBorder : createBorder;
  return borderFn({
    direction,
    outerBottomRight: css`
      ${({ theme }) => theme.colors.borderDarkest}
    `,
    innerBottomRight: css`
      ${({ theme }) => theme.colors.borderDark}
    `,
    outerTopLeft: css`
      ${({ theme }) => theme.colors.borderLightest}
    `,
    innerTopLeft: css`
      ${({ theme }) => theme.colors.borderLight}
    `,
  });
};

export const windowBorder = ({
  direction = 'extrude',
  useBoxShadow = false,
}: IBorderStyle = {}) => {
  let borderFn = useBoxShadow ? createBoxShadowBorder : createBorder;
  return borderFn({
    direction,
    outerBottomRight: css`
      ${({ theme }) => theme.colors.borderDarkest}
    `,
    innerBottomRight: css`
      ${({ theme }) => theme.colors.borderDark}
    `,
    outerTopLeft: css`
      ${({ theme }) => theme.colors.borderLight}
    `,
    innerTopLeft: css`
      ${({ theme }) => theme.colors.borderLightest}
    `,
  });
};

export const scrollBarBorder = ({
  direction = 'extrude',
  useBoxShadow = false,
}: IBorderStyle = {}) => {
  let borderFn = useBoxShadow ? createBoxShadowBorder : createBorder;
  return borderFn({
    direction,
    outerBottomRight: css`
      ${({ theme }) => theme.colors.borderDarkest}
    `,
    innerBottomRight: css`
      ${({ theme }) => theme.colors.borderDark}
    `,
    outerTopLeft: css`
      ${({ theme }) => theme.colors.borderLight}
    `,
    innerTopLeft: css`
      ${({ theme }) => theme.colors.borderLightest}
    `,
  });
};

export default border;
