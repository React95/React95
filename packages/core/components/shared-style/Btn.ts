import styled from '@xstyled/styled-components';
import { th } from '@xstyled/system';

export default styled.button`
  background-color: bg;
  padding: 7 20 5;
  border: 1px solid red;

  font-size: 12px;

  box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
    inset 0 0 0 1px ${th('colors.grays.3')}, 1px 1px 0 0px ${th('colors.black')};

  &:disabled {
    color: grays.3;
  }

  &:focus {
    outline: ${th('space.1')}px dotted ${th('colors.black')};
    outline-offset: -${th('space.5')}px;

    box-shadow: inset 1px 1px 0px 1px ${th('colors.white')},
      inset -0.5px -0.5px 0px 1px ${th('colors.grays.3')},
      1px 1px 0 1px ${th('colors.black')};
  }

  &:active {
    padding: 8 20 4;

    outline: ${th('space.1')}px dotted ${th('colors.black')};
    outline-offset: -${th('space.5')}px;

    box-shadow: inset 0 0 0 1px ${th('colors.grays.3')},
      0 0 0 1px ${th('colors.black')};
  }
`;
