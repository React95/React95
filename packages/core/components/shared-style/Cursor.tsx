import { css } from 'styled-components';
import auto from '../shared/assets/cursors/Arrow.png';
import text from '../shared/assets/cursors/Text.png';
import verticalText from '../shared/assets/cursors/VText.png';
import help from '../shared/assets/cursors/Help.png';
import crosshair from '../shared/assets/cursors/Cross.png';
import pointer from '../shared/assets/cursors/HandPointer.png';
import progress from '../shared/assets/cursors/AppStarting.png';
import wait from '../shared/assets/cursors/Wait.png';
import alias from '../shared/assets/cursors/Link.png';
import copy from '../shared/assets/cursors/Copy.png';
import move from '../shared/assets/cursors/Move.png';
import noDrop from '../shared/assets/cursors/Circle.png';
import notAllowed from '../shared/assets/cursors/NO.png';
import grab from '../shared/assets/cursors/Hand.png';
import grabbing from '../shared/assets/cursors/Handsqueezed.png';
import colResize from '../shared/assets/cursors/ColRezise.png';
import rowResize from '../shared/assets/cursors/RowResize.png';
import nResize from '../shared/assets/cursors/UpArrow.png';
import eResize from '../shared/assets/cursors/RightArrow.png';
import sResize from '../shared/assets/cursors/DownArrow.png';
import wResize from '../shared/assets/cursors/LeftArrow.png';
import nsResize from '../shared/assets/cursors/UpDownArrow.png';
import ewResize from '../shared/assets/cursors/LeftRight.png';
import neResize from '../shared/assets/cursors/UpRightArrow.png';
import nwResize from '../shared/assets/cursors/UpLeftArrow.png';
import seResize from '../shared/assets/cursors/DownRightArrow.png';
import swResize from '../shared/assets/cursors/DownLeftArrow.png';
import neswResize from '../shared/assets/cursors/AngleUpRight.png';
import nwseResize from '../shared/assets/cursors/AngleUpLeft.png';
import zoomIn from '../shared/assets/cursors/ZoomIn.png';
import zoomOut from '../shared/assets/cursors/ZoomOut.png';

const cursors = {
  auto,
  text,
  verticalText,
  help,
  crosshair,
  pointer,
  progress,
  wait,
  alias,
  copy,
  move,
  noDrop,
  notAllowed,
  grab,
  grabbing,
  colResize,
  rowResize,
  nResize,
  eResize,
  sResize,
  wResize,
  nsResize,
  ewResize,
  neResize,
  nwResize,
  seResize,
  swResize,
  neswResize,
  nwseResize,
  zoomIn,
  zoomOut,
};

export type ICursor =
  | 'auto'
  | 'default'
  | 'none'
  | 'context-menu'
  | 'help'
  | 'pointer'
  | 'progress'
  | 'wait'
  | 'crosshair'
  | 'text'
  | 'vertical-text'
  | 'alias'
  | 'copy'
  | 'move'
  | 'no-drop'
  | 'not-allowed'
  | 'grab'
  | 'grabbing'
  | 'col-resize'
  | 'row-resize'
  | 'n-resize'
  | 'e-resize'
  | 's-resize'
  | 'w-resize'
  | 'ns-resize'
  | 'ew-resize'
  | 'ne-resize'
  | 'nw-resize'
  | 'se-resize'
  | 'sw-resize'
  | 'nesw-resize'
  | 'nwse-resize'
  | 'zoom-in'
  | 'zoom-out';

const toCamelCase = (string: string) =>
  string.replace(/-./g, x => x.toUpperCase()[1]);

const Cursor = (cursor: ICursor) => {
  const camelCaseCursor = toCamelCase(cursor);
  const selectedCursor =
    cursors[camelCaseCursor] != null
      ? cursors[camelCaseCursor]
      : cursors['auto'];
  const cursorCss = css`
    cursor: url('${selectedCursor}'), ${cursor};
  `;
  const noneCss = css`
    cursor: none;
  `;
  console.log(cursor, camelCaseCursor);
  return cursor == 'none' ? noneCss : cursorCss;
};

export default Cursor;
