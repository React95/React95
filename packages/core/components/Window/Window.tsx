import styled from 'styled-components';
import { windowBorder } from './../shared-style/Border';
import CSS from 'csstype';

type pxInput =
  | number
  | CSS.Property.Width
  | CSS.Property.Height
  | CSS.Property.MinHeight
  | CSS.Property.MinWidth;
interface WindowProps {
  width?: CSS.Property.Width | number;
  height?: CSS.Property.Height | number;
  minHeight?: CSS.Property.MinHeight | number;
  minWidth?: CSS.Property.MinWidth | number;
  direction?: CSS.Property.FlexDirection;
}

const isNumeric = (input: string | number) => {
  return typeof input == 'string' && !isNaN(parseFloat(input));
};

const toPxUnits = (input: pxInput | undefined) => {
  return input && isNumeric(input.toString()) ? `${input}px` : input;
};

const Window = styled.section<WindowProps>`
  ${windowBorder()}
  background: ${({ theme }) => theme.colors.material};
  color: ${({ theme }) => theme.colors.materialText};
  width: ${({ width }) => toPxUnits(width)};
  min-width: ${({ minWidth }) => minWidth};
  height: ${({ height }) => toPxUnits(height)};
  min-height: ${({ minHeight }) => minHeight};
  padding: 3px;
  display: flex;
  flex-direction: ${({ direction }) => direction};;
`;
Window.defaultProps = {
  width: 320,
  height: 'auto',
  direction: 'column',
};

export default Window;
