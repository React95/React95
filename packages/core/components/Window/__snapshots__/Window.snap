// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Window /> Snapshot should match snapshot 1`] = `
.c0 {
  border: 0;
  box-shadow: inset -1px -1px 0 0  #0a0a0a , inset 1px 1px 0 0  #dfdfdf , inset -2px -2px 0 0  #848584 , inset 2px 2px 0 0  #fefefe;
  background: #c6c6c6;
  color: #0a0a0a;
  width: 320px;
  height: auto;
  padding: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <section
    class="c0"
    direction="column"
    height="auto"
    width="320"
  />
</div>
`;
